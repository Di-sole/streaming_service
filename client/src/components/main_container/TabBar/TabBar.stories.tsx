import type { Meta, StoryObj } from '@storybook/react';
import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';
import tabBar from './TabBar';
import './TabBar.module.scss';

const meta: Meta<typeof tabBar> = {
    title: 'TabBar',
    component: tabBar,
};

export default meta;

type Story = StoryObj<typeof tabBar>;

export const TabBar: Story = {
    parameters: {
        viewport: {
            viewports: INITIAL_VIEWPORTS,
            defaultViewport: 'ipad',
        },
    },
};
