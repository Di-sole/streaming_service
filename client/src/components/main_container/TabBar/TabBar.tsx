import { FC } from 'react';
import styles from './TabBar.module.scss';
import TabBarLinkUI from '../../UI/links/TabBarLink/TabBarLinkUI';
import homeIcon from '../../../../public/icons/tabbar/home.svg';
import catalogueIcon from '../../../../public/icons/tabbar/catalogue.svg';
import magnifyingGlass from '../../../../public/icons/tabbar/magnifying_glass.svg';
import monitorIcon from '../../../../public/icons/tabbar/monitor.svg';
import dotsIcon from '../../../../public/icons/tabbar/dots.svg';
import { useRouter } from 'next/router';

const TabBar: FC = () => {
    const { pathname, asPath, query } = useRouter();

    return (
        <div className={styles.container} data-testid={'tabBar'}>
            <div className={styles.container__tabBar}>
                <TabBarLinkUI
                    selected={asPath === '/' ? true : false}
                    href="https://www.ivi.ru/"
                    icon={homeIcon}
                    text="Мой Иви"
                />
                <TabBarLinkUI
                    selected={asPath === '/movies' ? true : false}
                    href="/movies"
                    icon={catalogueIcon}
                    text="Каталог"
                />
                <TabBarLinkUI
                    selected={'ivi_search' in query ? true : false}
                    href={`${asPath}?ivi_search`}
                    icon={magnifyingGlass}
                    text="Поиск"
                />
                <TabBarLinkUI
                    selected={pathname === '/tvplus' ? true : false}
                    href="https://www.ivi.ru/tvplus"
                    icon={monitorIcon}
                    text="TV+"
                />
                <TabBarLinkUI selected={pathname === '/more' ? true : false} href="#!" icon={dotsIcon} text="Ещё" />
            </div>
        </div>
    );
};

export default TabBar;
