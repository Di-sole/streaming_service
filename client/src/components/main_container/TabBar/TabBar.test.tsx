import { render, screen } from '@testing-library/react';
import TabBar from './TabBar';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            route: '/',
            pathname: '',
            asPath: '/?ivi_search',
            query: { keyword: 'ivi_search' },
        };
    },
}));

describe('TABBAR SNAPSHOTS TESTS', () => {
    test('TabBar should not be changed', () => {
        render(<TabBar />);

        const div = screen.getByTestId('tabBar');
        expect(div).toMatchSnapshot();
    });
});
