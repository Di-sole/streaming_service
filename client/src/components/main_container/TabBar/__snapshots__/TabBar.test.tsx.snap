// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TABBAR SNAPSHOTS TESTS TabBar should not be changed 1`] = `
<div
  class="container"
  data-testid="tabBar"
>
  <div
    class="container__tabBar"
  >
    <a
      class="container "
      href="https://www.ivi.ru/"
      id="tab-item"
    >
      <div
        class="container__itemGlow"
      />
      <div
        class="container__itemIcon"
      >
        <img
          alt="Иконка Мой Иви"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
      </div>
      <div
        class="container__itemCaption"
      >
        Мой Иви
      </div>
    </a>
    <a
      class="container "
      href="/movies"
      id="tab-item"
    >
      <div
        class="container__itemGlow"
      />
      <div
        class="container__itemIcon"
      >
        <img
          alt="Иконка Каталог"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
      </div>
      <div
        class="container__itemCaption"
      >
        Каталог
      </div>
    </a>
    <a
      class="container "
      href="/?ivi_search?ivi_search"
      id="tab-item"
    >
      <div
        class="container__itemGlow"
      />
      <div
        class="container__itemIcon"
      >
        <img
          alt="Иконка Поиск"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
      </div>
      <div
        class="container__itemCaption"
      >
        Поиск
      </div>
    </a>
    <a
      class="container "
      href="https://www.ivi.ru/tvplus"
      id="tab-item"
    >
      <div
        class="container__itemGlow"
      />
      <div
        class="container__itemIcon"
      >
        <img
          alt="Иконка TV+"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
      </div>
      <div
        class="container__itemCaption"
      >
        TV+
      </div>
    </a>
    <a
      class="container "
      href="#!"
      id="tab-item"
    >
      <div
        class="container__itemGlow"
      />
      <div
        class="container__itemIcon"
      >
        <img
          alt="Иконка Ещё"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
      </div>
      <div
        class="container__itemCaption"
      >
        Ещё
      </div>
    </a>
  </div>
</div>
`;
